### PR TITLE
Add warnings

### PR DESCRIPTION
**Pull request by GabrielRavier**
This adds the `WARNINGS` (`-Wall -Wextra -pedantic`), `ALL_WARNINGS` (`-Weverything`) and the `FATAL_ERRORS` (`-Werror`) options. These are portable across MSVC, ICC, Clang and GCC and will just show a warning on unsupported configurations (such as `ALL_WARNINGS` on non-MSVC/clang or using a compiler that's not MSVC/ICC/Clang/GCC). This is only restricted to those compilers due to CMake not supporting any way of portably adding warnings (yet).

I also added these to the `Makefile`.

### CODE REVIEW

**Comment by Clownacy** [comment]
Are you sure it should affect global CMake variables? Wouldn't that mean we'd get warnings from the dependencies as well?

**Comment by GabrielRavier** [comment]
True, I'll change it to only affect our stuff

**Comment by GabrielRavier** [comment]
Fixed, after only 24 days :stuck_out_tongue:

File: CMakeLists.txt [hunk 1]
```diff
@@ -20,6 +47,66 @@ if(MSVC)
...
+
+		# Can't do this with target_compile_options
+		# if(CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
+		# 	string(REGEX REPLACE "/W[0-4]" "/W4" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+		# else()
+		# 	target_compile_options(CSE2 PRIVATE /W4)
+		# endif()
+
+		target_compile_options(CSE2 PRIVATE /W4)
+		target_compile_options(bin2h PRIVATE /W4)
```
Comment (Clownacy) [inline CMakeLists.txt]: I think the recent changes to bin2h make lines this useless.

Comment (GabrielRavier) [reply CMakeLists.txt]: What do you mean ?

Comment (Clownacy) [reply CMakeLists.txt]: bin2h has been split to a separate CMake project, but in a way that makes it work with cross-compilation. The way it's done means the bin2h target is now just an executable.

Comment (GabrielRavier) [reply CMakeLists.txt]: I'll check if this actually makes this useless, and will fix it in this case

**Comment by GabrielRavier** [comment]
Closing this, will make a better version of this soon

**Pull request closed without merging**